Compile error when compiling with -Werror using gcc 12.0.1

## System information
Cross compiling media-driver 22.2.1 using gcc 12-20220220

## Issue behavior
### Describe the current behavior
Fails compile with:
- `warning: dangling pointer 'address2Counter' to 'ctr' may be used [-Wdangling-pointer=]`
- `warning: argument 1 range [18446744071562067968, 18446744073709551615] exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]`

### Describe the expected behavior
Compiles correctly with gcc 11.2.0
Using the following compile options allows these errors to pass as warnings:
- `export CXXFLAGS="${CXXFLAGS} -Wno-error=alloc-size-larger-than= -Wno-error=dangling-pointer"`

## build log

```
[43/942] Building CXX object media_driver/CMakeFiles/iHD_drv_video_OBJ.dir/agnostic/common/cm/cm_kernel_ex.cpp.o
In file included from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_common/agnostic/common/os/mos_utilities.h:138,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_common/agnostic/common/os/mos_os.h:32,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/os/mos_context.h:30,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/linux/common/ddi/media_libva_common.h:37,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/linux/common/cm/ddi/media_libva_cm.h:30,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/linux/common/cm/hal/cm_def_os.h:41,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/cm/cm_def.h:30,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/cm/cm_kernel.h:30,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/cm/cm_kernel_rt.h:30,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/cm/cm_kernel_ex.h:28,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/cm/cm_kernel_ex.cpp:27:
In static member function 'static _Ty* MosUtilities::MosNewArrayUtil(int32_t) [with _Ty = unsigned char; _Types = {}]',
    inlined from '_Ty* MOS_NewArrayUtil(int32_t) [with _Ty = unsigned char; _Types = {}]' at /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_common/agnostic/common/os/mos_utilities.h:185:46,
    inlined from 'virtual int32_t CmKernelEx::Initialize(const char*, const char*)' at /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/cm/cm_kernel_ex.cpp:192:22,
    inlined from 'virtual int32_t CmKernelEx::Initialize(const char*, const char*)' at /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/cm/cm_kernel_ex.cpp:70:9:
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_softlet/agnostic/common/os/mos_utilities_next.h:106:20: warning: argument 1 range [18446744071562067968, 18446744073709551615] exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
  106 |         _Ty* ptr = new (std::nothrow) _Ty[numElements]();
      |                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/toolchain/x86_64-libreelec-linux-gnu/include/c++/12.0.1/bits/exception_ptr.h:40,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/toolchain/x86_64-libreelec-linux-gnu/include/c++/12.0.1/exception:168,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/toolchain/x86_64-libreelec-linux-gnu/include/c++/12.0.1/ios:39,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/toolchain/x86_64-libreelec-linux-gnu/include/c++/12.0.1/ostream:38,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/toolchain/x86_64-libreelec-linux-gnu/include/c++/12.0.1/iostream:39,
                 from /var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/linux/common/cm/hal/cm_def_os.h:34:
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/toolchain/x86_64-libreelec-linux-gnu/include/c++/12.0.1/new: In member function 'virtual int32_t CmKernelEx::Initialize(const char*, const char*)':
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/toolchain/x86_64-libreelec-linux-gnu/include/c++/12.0.1/new:142:26: note: in a call to allocation function 'void* operator new [](std::size_t, const std::nothrow_t&)' declared here
  142 | _GLIBCXX_NODISCARD void* operator new[](std::size_t, const std::nothrow_t&) _GLIBCXX_USE_NOEXCEPT
      |                          ^~~~~~~~
[95/942] Building CXX object media_driver/CMakeFiles/iHD_drv_video_OBJ.dir/agnostic/common/codec/hal/codechal_encoder_base.cpp.o
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/codec/hal/codechal_encoder_base.cpp: In member function 'MOS_STATUS CodechalEncoderState::ReadCounterValue(uint16_t, EncodeStatusReport*)':
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/codec/hal/codechal_encoder_base.cpp:3941:43: warning: dangling pointer 'address2Counter' to 'ctr' may be used [-Wdangling-pointer=]
 3941 |     encodeStatusReport->HWCounterValue.IV = *(++address2Counter);
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/codec/hal/codechal_encoder_base.cpp:3926:18: note: 'ctr' declared here
 3926 |         uint32_t ctr[4] = { 0 };
      |                  ^~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/codec/hal/codechal_encoder_base.cpp:3937:48: warning: dangling pointer 'address2Counter' to 'ctr' may be used [-Wdangling-pointer=]
 3937 |     encodeStatusReport->HWCounterValue.Count = *address2Counter;
      |                                                ^~~~~~~~~~~~~~~~
/var/media/DATA/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-11.0-devel/build/media-driver-22.2.1/media_driver/agnostic/common/codec/hal/codechal_encoder_base.cpp:3926:18: note: 'ctr' declared here
 3926 |         uint32_t ctr[4] = { 0 };
      |                  ^~~
```

--- a/media_common/agnostic/common/os/mos_utilities.h
+++ b/media_common/agnostic/common/os/mos_utilities.h
@@ -181,10 +181,10 @@ _Ty* MOS_NewUtil(_Types&&... _Args)
 template<class _Ty, class... _Types>
 _Ty *MOS_NewArrayUtil(const char *functionName,
     const char *filename,
-    int32_t line, int32_t numElements)
+    int32_t line, size_t numElements)
 #else
 template<class _Ty, class... _Types>
-_Ty* MOS_NewArrayUtil(int32_t numElements)
+_Ty* MOS_NewArrayUtil(size_t numElements)
 #endif
 {
 #if MOS_MESSAGES_ENABLED

--- a/media_driver/agnostic/common/codec/hal/codechal_encoder_base.cpp
+++ b/media_driver/agnostic/common/codec/hal/codechal_encoder_base.cpp
@@ -3891,6 +3891,7 @@ MOS_STATUS CodechalEncoderState::ReadCounterValue(uint16_t index, EncodeStatusRe
     CODECHAL_ENCODE_FUNCTION_ENTER;
     CODECHAL_ENCODE_CHK_NULL_RETURN(encodeStatusReport);
     uint64_t *address2Counter = nullptr;
+    uint32_t ctr[4] = { 0 };
 
     if (m_hwInterface->GetCpInterface()->IsHwCounterIncrement(m_osInterface))
     {
@@ -3923,7 +3924,6 @@ MOS_STATUS CodechalEncoderState::ReadCounterValue(uint16_t index, EncodeStatusRe
     else
     {
         //Report driver generated counter which was submitted to HW by command
-        uint32_t ctr[4] = { 0 };
         eStatus = m_hwInterface->GetCpInterface()->GetCounterValue(ctr);
         if (MOS_STATUS_SUCCESS == eStatus)
         {

--- a/media_driver/linux/common/codec/ddi/media_ddi_decode_vp8.cpp
+++ b/media_driver/linux/common/codec/ddi/media_ddi_decode_vp8.cpp
@@ -235,20 +235,17 @@ VAStatus DdiDecodeVP8::ParsePicParams(
     codecPicParams->ucUvModeProbs[0]            = picParam->uv_mode_probs[0];
     codecPicParams->ucUvModeProbs[1]            = picParam->uv_mode_probs[1];
     codecPicParams->ucUvModeProbs[2]            = picParam->uv_mode_probs[2];
-    if (codecPicParams->ucMvUpdateProb[0] && picParam->mv_probs[0])
-    {
-        MOS_SecureMemcpy(codecPicParams->ucMvUpdateProb[0],
-            sizeof(codecPicParams->ucMvUpdateProb[0]),
-            picParam->mv_probs[0],
-            sizeof(codecPicParams->ucMvUpdateProb[0]));
-    }
-    if (codecPicParams->ucMvUpdateProb[1] && picParam->mv_probs[1])
-    {
-        MOS_SecureMemcpy(codecPicParams->ucMvUpdateProb[1],
-            sizeof(codecPicParams->ucMvUpdateProb[1]),
-            picParam->mv_probs[1],
-            sizeof(codecPicParams->ucMvUpdateProb[1]));
-    }
+
+    MOS_SecureMemcpy(codecPicParams->ucMvUpdateProb[0],
+        sizeof(codecPicParams->ucMvUpdateProb[0]),
+        picParam->mv_probs[0],
+        sizeof(codecPicParams->ucMvUpdateProb[0]));
+
+    MOS_SecureMemcpy(codecPicParams->ucMvUpdateProb[1],
+        sizeof(codecPicParams->ucMvUpdateProb[1]),
+        picParam->mv_probs[1],
+        sizeof(codecPicParams->ucMvUpdateProb[1]));
+
     codecPicParams->ucP0EntropyCount = (8 - picParam->bool_coder_ctx.count) & 0x7;  //hardware needs used bits not remaining bits in bool decoder
     codecPicParams->ucP0EntropyValue = picParam->bool_coder_ctx.value;
     codecPicParams->uiP0EntropyRange = picParam->bool_coder_ctx.range;

--- a/media_softlet/agnostic/common/os/mos_utilities_next.h
+++ b/media_softlet/agnostic/common/os/mos_utilities_next.h
@@ -90,10 +90,10 @@ public:
     template<class _Ty, class... _Types>
     static _Ty *MosNewArrayUtil(const char *functionName,
         const char *filename,
-        int32_t line, int32_t numElements)
+        int32_t line, size_t numElements)
 #else
     template<class _Ty, class... _Types>
-    static _Ty* MosNewArrayUtil(int32_t numElements)
+    static _Ty* MosNewArrayUtil(size_t numElements)
 #endif
     {
 #if (_DEBUG || _RELEASE_INTERNAL)

